pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -774,7 +796,37 @@ func (r *ReconcilePerconaServerMongoDB) reconcileMongodConfigMaps(cr *api.Percon
 			},
 		})
 		if err != nil {
-			return err
+			return errors.Wrap(err, "create or update config map")
+		}
+
+		if !rs.NonVoting.Enabled {
+			continue
+		}
+
+		name = psmdb.MongodCustomConfigName(cr.Name, rs.Name+"-nv")
+		if rs.NonVoting.Configuration == "" {
+			if err := deleteConfigMapIfExists(r.client, cr, name); err != nil {
+				return errors.Wrap(err, "failed to delete nonvoting mongod config map")
+			}
+
+			continue
+		}
+
+		err = r.createOrUpdateConfigMap(cr, &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{
```

Comment:
as i remember, you don't need to fill `TypeMeta`

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/723/commits/e7ccc8c0b0490d529dcb363494a682bc17a95b1c